sys/fido2/ctap/ctap.c

```diff
@@ -0,0 +1,1866 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup fido2_ctap
+ * @{
+ * @file
+ *
+ * @author  Nils Ollrogge <nils.ollrogge@fu-berlin.de>
+ * @}
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <assert.h>
+
+#include "errno.h"
+#include "fmt.h"
+#include "xtimer.h"
+#include "byteorder.h"
+
+#include "fido2/ctap/transport/ctap_transport.h"
+#include "fido2/ctap.h"
+#include "fido2/ctap/ctap_utils.h"
+#include "fido2/ctap/ctap_cbor.h"
+#include "fido2/ctap/ctap_mem.h"
+
+#if IS_USED(MODULE_FIDO2_CTAP_HID)
+#include "fido2/ctap/transport/hid/ctap_hid.h"
+#endif
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/**
+ * @brief CTAP authenticator config struct
+ */
+typedef struct {
+    uint8_t options;                    /**< options */
+    uint8_t aaguid[CTAP_AAGUID_SIZE];   /**< AAGUID of device */
+} ctap_config_t;
+
+/**
+ * @brief CTAP state struct
+ *
+ * state of authenticator. Stored in flash memory
+ */
+typedef struct {
+    uint8_t initialized_marker;                 /**< CTAP initialized marker */
+    bool pin_is_set;                            /**< PIN is set or not */
+    int rem_pin_att;                            /**< remaining PIN tries */
+    uint8_t pin_hash[SHA256_DIGEST_LENGTH / 2]; /**< LEFT(SHA-256(pin), 16) */
```

Comment:
Here and in other occasions: Didn't you introduce a `define` for these sizes? If yes, please use it. If no, leave it as is.

Reply:
Not for this size. Only for the HMAC-SHA256 ones. I think using `SHA256_DIGEST_LENGTH / 2` is better though ? Maybe I just remove the other defines.